k8s/plugin/src/resources/upgrade.rs

```diff
@@ -0,0 +1,430 @@
+                    }
+                    Err(e) => {
+                        println!("Failed in fetching deployment {:?}", e);
+                        std::process::exit(1);
+                    }
+                }
+
+                // Create Service
+                match uo.get_service().await {
+                    Ok(svc) => {
+                        if svc.iter().count() == 0 {
+                            let ns = Some(ns.to_string());
+                            let service = objects::upgrade_operator_service(ns);
+                            match uo.service.create(&pp, &service).await {
+                                Ok(svc) => {
+                                    println!(
+                                        "Service : {} created in namespace : {}.",
+                                        svc.metadata.name.unwrap(),
+                                        svc.metadata.namespace.unwrap(),
+                                    );
+                                }
+                                Err(e) => {
+                                    println!("Failed in creating service {:?}", e);
+                                    tokio::time::sleep(Duration::from_secs(1)).await;
+                                    std::process::exit(1);
+                                }
+                            }
+                        } else {
+                            println!(
+                                "Service : {} in namespace : {} already exist.",
+                                svc.items[0].metadata.name.as_ref().unwrap(),
+                                svc.items[0].metadata.namespace.as_ref().unwrap(),
+                            );
+                        }
+                    }
+                    Err(e) => {
+                        println!("Failed in fetching service {:?}", e);
+                        std::process::exit(1);
+                    }
+                }
+            }
+            Err(e) => println!("Failed to install. Error {:?}", e),
+        };
+    }
+
+    /// Uninstall the upgrade resources
+    pub async fn uninstall(ns: &str) {
+        match UpgradeResources::new(ns).await {
+            Ok(duo) => {
+                let dp = DeleteParams::default();
+
+                // delete deployment
+                match duo.get_deployment().await {
+                    Ok(deployment) => {
+                        if deployment.iter().count() == 1 {
+                            match duo
+                                .deployment
+                                .delete(UPGRADE_CONTROLLER_DEPLOYMENT, &dp)
+                                .await
+                            {
+                                Ok(_) => {
+                                    // wait unit the deployment is deleted.
+                                    while duo.get_deployment().await.iter().count() != 0 {
+                                        tokio::time::sleep(Duration::from_secs(5)).await;
```

Comment:
Should we add a timeout/max retries?
Alternatively, do we even need to wait for this on the install? Shouldn't the install itself wait?